orbit/cmd/desktop/desktop.go

```diff
@@ -83,77 +48,84 @@ func main() {
 		myDeviceItem.Disable()
 		transparencyItem := systray.AddMenuItem("Transparency", "")
 
+		var insecureSkipVerify bool
+		if os.Getenv("FLEET_DESKTOP_INSECURE") != "" {
+			insecureSkipVerify = true
+		}
+
+		// TODO: figure out the right rootCA to pass to the client
+		client, err := service.NewDeviceClient(basePath, insecureSkipVerify, "")
+
+		if err != nil {
+			log.Printf("unable to initialize request client: %s", err)
+			os.Exit(1)
+		}
+
 		// Perform API test call to enable the "My device" item as soon
 		// as the device auth token is registered by Fleet.
-		go func() {
-			ticker := time.NewTicker(5 * time.Second)
-			defer ticker.Stop()
+		deviceEnabledChan := func() <-chan bool {
+			done := make(chan bool)
```

Comment:
Looks good.

For these type of channels that are only used to signal a one-off "done", one typically uses `chan struct{}` and then instead of writing to it, you just do something like `defer close(done)` in the inner function.

Reply:
neat! I will update with that, thank you!